torch/nn/functional.py

```diff
@@ -4990,6 +4990,229 @@ def multi_head_attention_forward(
           :math:`S` is the source sequence length. If ``average_attn_weights=False``, returns attention weights per
           head of shape :math:`(num_heads, L, S)` when input is unbatched or :math:`(N, num_heads, L, S)`.
     """
+
```

Comment:
Is this whole code completely duplicated from the nn.Module implementation? That does not sound right. We should either need a single implementation of this or refactor it into a helper function.